recipes/newrelic/infrastructure/ohi/nginx/debian.yml

```diff
@@ -35,82 +35,107 @@ logMatch:
     attributes:
       logtype: nginx-error
 
-validationNrql: "SELECT count(*) from NginxSample where hostname like '{{.HOSTNAME}}' FACET entityGuid SINCE 10 minutes ago"
+validationNrql: "SELECT count(*) from NginxSample where hostname like '{{.HOSTNAME}}%' FACET entityGuid SINCE 10 minutes ago"
 
 successLinkConfig:
   type: EXPLORER
 
-inputVars:
-  - name: "NR_CLI_STUB_STATUS_URL"
-    prompt: "NGINX status URL (default: http://127.0.0.1/status)"
-    default: "http://127.0.0.1/status"
-
 preInstall:
   info: |2
       To capture data from the NGINX integration, you'll first need to enable
       and configure the applicable extension module:
       - For NGINX Open Source: HTTP stub status module 
       - For NGINX Plus: HTTP status module and HTTP API module
+      - See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/nginx-monitoring-integration#config for more info.
 
 install:
   version: "3"
   silent: true
-
   tasks:
     default:
       cmds:
-        - task: assert_pre_req
-        - task: setup
+        - task: assert_infra
+        - task: input_assert
+          vars:
+            MAX_RETRIES: 3
         - task: restart
 
-    assert_pre_req:
+    assert_infra:
       cmds:
         - |
           IS_INFRA_INSTALLED=$(sudo ps aux | grep newrelic-infra-service | grep -v grep | wc -l)
           if [ $IS_INFRA_INSTALLED -eq 0 ] ; then
             echo "The infrastructure agent is required to install this integration, we recommend going through our guided install path for this pre-requisite which can be found here:  https://docs.newrelic.com/docs/new-relic-guided-installation-overview" >> /dev/stderr
+            sudo rm /tmp/nginx-ohai-input
             exit 1
           fi
+
+    input_assert:
+      cmds:
         - |
-          STUB_STATUS_ENABLED=$(curl {{.NR_CLI_STUB_STATUS_URL}} -s | grep "Active connections:" | wc -l)
-          if [ $STUB_STATUS_ENABLED -eq 0 ] ; then
-            echo "NGINX HTTP status module must be configured for New Relic to collect data. \n - See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/nginx-monitoring-integration#config for more info." >> /dev/stderr
+          tries=0
+
+          # Check Env Vars
+          NR_CLI_STUB_STATUS_URL="{{.NR_CLI_STUB_STATUS_URL}}"
+          if [[ "$NR_CLI_STUB_STATUS_URL" == "" ]]; then
+            while [ $tries -lt {{.MAX_RETRIES}} ]; do
+              printf "\nNGINX status URL (default: http://127.0.0.1/status): "
```

Comment:
If the install is run with assumeYes (-y) which is this go-task variable {{.NEW_RELIC_ASSUME_YES}} , then we should not prompt for the URL and use the default value of http://127.0.0.1/status

One step further perhaps, and independently of assumeYes, maybe we could try to see if the default URL works, and only prompt if we see it doesn't work?

Reply:
@Julien4218 

Regarding AssumeYes, we are reading the env vars, which I assume are being injected by the test unit, and if empty we proceed with user input. Is this approach ok or should we use {{.NEW_RELIC_ASSUME_YES}}?

As for your second suggestion, that's a good idea. We'll keep that in mind for a follow up review of this recipe.
Thanks!

Reply:
Yes, what you've done with the reading of the env vars is great, this mainly supports our test framework.
However customer would rely on the AssumeYes option to essentially say "just install with all defaults", and some customers may potentially inject the specific values through env var (just like our test) but this would be optional (maybe different value than default).
So in short, we should support both, and always prompt if there is no env var value or no default value.

Reply:
Understood now @Julien4218 . Thanks for the details.

I'll change that and resubmit.
Thanks!